twine/commands/upload.py

```diff
@@ -64,6 +64,11 @@ def upload(upload_settings: settings.Settings, dists: List[str]) -> None:
     repository = upload_settings.create_repository()
     uploaded_packages = []
 
+    if upload_settings.verbose:
+        print("\nList of packages that will be uploaded:")
+        for package in uploads:
+            print("  " + os.path.basename(package))
+        print("\n")
```

Comment:
Seeing your screenshot makes me think this could be more useful, e.g.:

```
$ twine upload -r testpypi --verbose dist/* 
Uploading distributions to https://test.pypi.org/legacy/
  dist/example_pkg_bhrutledge-0.0.4a4-py3-none-any.whl (2.6 KB)
  dist/example-pkg-bhrutledge-0.0.4a3.tar.gz (1.3 KB)
  dist/example-pkg-bhrutledge-0.0.4a4.tar.gz (1.4 KB)

Enter your username: 
```

Specifically:

- Paired with the "Uploading distributions" line
- Showing the full path, instead of the file name
- Showing the file size (just an idea, could be done later in a separate PR)

Reply:
> Showing the file size (just an idea, could be done later in a separate PR)

I like this. I wonder if we can also show that we're uploading the associated `.asc` file too so people stop asking why we aren't uploading it (when we are)

Reply:
> I wonder if we can also show that we're uploading the associated `.asc` file too so people stop asking why we aren't uploading it (when we are)

I wondered the same thing, and suggested it in https://github.com/pypa/twine/issues/381#issuecomment-538103137. However, I thought that might be better handled in a separate PR, since it's a separate operation, later in `upload`:

https://github.com/pypa/twine/blob/31cad981425b2ef8ee17f74925320acc356ab3a6/twine/commands/upload.py#L82-L86

See this makes me wonder if `add_gpg_signature` would be the right spot for a verbose message.

Reply:
> Please fix your failing tests https://travis-ci.org/github/pypa/twine/builds/695224649 and clean up your commit history on the branch
> Paired with the "Uploading distributions" line
Showing the full path, instead of just the file name

Thanks for help! I will fix the tests, and make these changes in a PR. 

> Showing the file size (just an idea, could be done later in a separate PR)
> we can also show that we're uploading the associated .asc file too so people stop asking why we aren't uploading it (when we are)

I will create a checklist on #381 that includes these changes as well as the others discussed in #381.

Reply:
Sounds good, @VikramJayanthi17. If you're game, I think it would be okay to show the file size as part of this PR.

Reply:
> If you're game, I think it would be okay to show the file size as part of this PR.

Sounds good, I will add it to this PR.

Reply:
> See this makes me wonder if `add_gpg_signature` would be the right spot for a verbose message.

This makes me think we're printing this way too early if we can't rely on the PackageFile instance for each item. We can print the filename and the asc filename at that point

Reply:
@sigmavirus24 Can we hold on the signature discussion for now? It'll be implemented in a subsequent PR.

Reply:
@bhrutledge no, because the signatures should be printed in the same place and so that makes me believe that this is the wrong place to be printing this informatoin.